src/lib.rs

```diff
@@ -692,6 +746,54 @@ impl<R: ?Sized> Rng for Box<R> where R: Rng {
     }
 }
 
+/// Trait for casting types to byte slices
+pub trait AsByteSlice {
+    /// Return a mutable reference to self as a byte slice
+    fn as_byte_slice<'a>(&'a mut self) -> &'a mut [u8];
+    /// Perform byte-swapping on BE platforms
+    fn to_le(&mut self);
```

Comment:
This does not really need to be part of the trait. Wouldn't it be easier to just write the conversion loop in `fill` and `try_fill`?

Reply:
Okay, then we can also standardise the naming.

Reply:
It will require to use more complex trait bounds, not only you'll need trait for iteration, but also for `to_le` method as well.